skfda/preprocessing/smoothing/_basis.py

```diff
@@ -0,0 +1,520 @@
+        differential operator:
+
+        >>> from skfda.misc import LinearDifferentialOperator
+        >>> fd = skfda.FDataGrid(data_matrix=x, sample_points=t)
+        >>> basis = skfda.representation.basis.Fourier((0, 1), nbasis=3)
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='cholesky',
+        ...                smoothing_parameter=1,
+        ...                penalty=LinearDifferentialOperator(weights=[3, 5]),
+        ...                return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[ 0.18,  0.07,  0.09]])
+
+        >>> from skfda.misc import LinearDifferentialOperator
+        >>> fd = skfda.FDataGrid(data_matrix=x, sample_points=t)
+        >>> basis = skfda.representation.basis.Fourier((0, 1), nbasis=3)
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='qr',
+        ...                smoothing_parameter=1,
+        ...                penalty=LinearDifferentialOperator(weights=[3, 5]),
+        ...                return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[ 0.18,  0.07,  0.09]])
+
+        >>> from skfda.misc import LinearDifferentialOperator
+        >>> fd = skfda.FDataGrid(data_matrix=x, sample_points=t)
+        >>> basis = skfda.representation.basis.Fourier((0, 1), nbasis=3)
+        >>> smoother = skfda.preprocessing.smoothing.BasisSmoother(
+        ...                basis, method='matrix',
+        ...                smoothing_parameter=1,
+        ...                penalty=LinearDifferentialOperator(weights=[3, 5]),
+        ...                return_basis=True)
+        >>> fd_basis = smoother.fit_transform(fd)
+        >>> fd_basis.coefficients.round(2)
+        array([[ 0.18,  0.07,  0.09]])
+
+    References:
+        .. [RS05-5-2-5] Ramsay, J., Silverman, B. W. (2005). How spline
+            smooths are computed. In *Functional Data Analysis*
+            (pp. 86-87). Springer.
+
+        .. [RS05-5-2-7] Ramsay, J., Silverman, B. W. (2005). HSpline
+            smoothing as an augmented least squares problem. In *Functional
+            Data Analysis* (pp. 86-87). Springer.
+
+    """
+
+    _required_parameters = ["basis"]
+
+    class SolverMethod(Enum):
+        cholesky = _Cholesky()
+        qr = _QR()
+        matrix = _Matrix()
+
+    def __init__(self,
+                 basis,
+                 *,
+                 smoothing_parameter: float = 0,
+                 weights=None,
+                 penalty: Union[int, Iterable[float],
+                                'LinearDifferentialOperator'] = None,
+                 penalty_matrix=None,
+                 output_points=None,
+                 method='cholesky',
+                 keepdims=False,
```

Comment:
I think the ``keepdims`` parameter shouldn't be here. It should be discussed whether to eliminate the keepdims parameter from the ``FData``, leaving it only as an argument for the evaluate function. Anyway, ``keepdims`` should inherit the value of the object to be smoothed.

Reply:
`keepdims` usage eliminated in this module in 73ec1d07ea2d6aae469481a3ed30eaa426f4b273.